### PR TITLE
feat: implements count method and pagination with total

### PR DESCRIPTION
**Pull request by cofin**
This enhancements currently adds an `count` method to the `AbstractRepository` and `SQLAlchemyRepository` classes.  

It also implements another `PaginatedSQLAlchemyRepository` class that implements a `list` method returning a `tuple[list[ModelT], int]` instead of `list[ModelT]`.  This allows for easy integration with the Limit/Offset Pagination within Starlite.

I'm still trying to wrap my head around how the testing works, so it's lacking a bit there.

I've opened this PR up even though it's not quite ready so you can weigh in.  Let me know if you think there's a better way to implement this or it's something you aren't looking to include here.

### CODE REVIEW

File: src/starlite_saqlalchemy/service/sqlalchemy.py [hunk 1]
```diff
@@ -122,3 +122,24 @@ async def new(cls: type[RepoServiceT]) -> AsyncIterator[RepoServiceT]:
         """
         async with async_session_factory() as session:
             yield cls(session=session)
+
+
+class PaginatedRepositoryService(RepositoryService):
```
Comment (peterschutt) [inline src/starlite_saqlalchemy/service/sqlalchemy.py:127]: While we are still breaking things, I'd rather just change this on `RepositoryService`.

**Comment by peterschutt** [comment]
Awesome, thanks @cofin.

> I'm still trying to wrap my head around how the testing works, so it's lacking a bit there.

Tests are organised by required dependencies. For example, if SQLAlchemy isn't installed in the environment where the tests are being run, pytest won't collect any of the tests under the `require_sqlalchemy` namespace. 

If some piece of functionality requires a specific extra dependency, then put the tests for it under the `requires_<dependency>` namespace.

If some piece of functionality should behave the same, irrespective of available dependencies then those tests can go in the top-level of the tests/unit suite.

All of the test environments are configured in tox, most notable is `no-extras` env, which tests the app under minimal install conditions.

If you want to have all the tests run, have tox, python 3.10 and 3.11 available and run `tox`. This runs all the tests exactly as CI does.

If you are just iterating, run pytest at the command line and it will run the tests dependent on the extras available in your local environment.

LMK if I've missed anything, or you think I could structure those better.

This reminds me I  should update `CONTRIBUTING.md`.

**Comment by cofin** [comment]
There was an attempt to write test cases.  Please take a look to see if you think we should improve.

**Comment by peterschutt** [comment]
Thanks @coffin! closing in favor of #276

**Pull request closed without merging**